### PR TITLE
Build `ui` styles into `ui-react` package

### PR DESCRIPTION
**Pull request by reesscot**
*Issue #, if available:*
Fixes: https://github.com/aws-amplify/amplify-ui/issues/1020 and https://github.com/aws-amplify/amplify-ui/issues/1005

*Description of changes:*
Customers are experiencing an issue with the [parcel bundler](https://parceljs.org/recipes/react/) where the CSS statement `@import '@aws-amplify/ui/styles.css';` pointing to `ui` package cannot be resolved via `node_modules`. This PR fixes this by building the styles from `ui` into `ui-react` using `tsup`.

Testing TODO:

- [x] Test in ViteJS (Scott)
- [x] Test in Parcel (Scott)
- [x] Test in NextJS (Scott)
- [x] Test in CRA (Scott)
- [x] Test in UMJS (Erik)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ErikCH — APPROVED** [review]
👍

File: packages/react/src/styles.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import './styles.css';
```
Comment (dbanksdesign) [inline packages/react/src/styles.ts:1]: OMG this is way cleaner than writing a build script to copy the file over. Bravo.

File: packages/react/src/styles.css [hunk 2]
```diff
@@ -1 +1 @@
-@import '@aws-amplify/ui/styles.css';
+@import '../../ui/dist/styles.css';
```
Comment (dbanksdesign) [inline packages/react/src/styles.css:1]: If we are using tsup for the build script, do we still need to change this line? Just curious

**Review by dbanksdesign — APPROVED** [review]
LGTM! Just one question for my own ignorance.

File: packages/react/src/styles.css (hunk 2 above)
Comment (reesscot) [reply packages/react/src/styles.css:1]: Unfortunately, no. If I don't change the path to point to the relative file, `tsup` will just copy it the import statement:
```
@import"@aws-amplify/ui/styles.css";
```
So, for `tsup` to copy the file, we must use the relative path.

**Pull request merged**